cmd/snap-update-ns/mount-entry.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SNAP_CONFINE_MOUNT_ENTRY_H
+#define SNAP_CONFINE_MOUNT_ENTRY_H
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+/**
+ * A fstab-like mount entry.
+ **/
+struct sc_mount_entry {
+	char *mnt_fsname;	/* name of mounted filesystem */
+	char *mnt_dir;		/* filesystem path prefix */
+	char *mnt_type;		/* mount type (see mntent.h) */
+	char *mnt_opts;		/* mount options (see mntent.h) */
+	int mnt_freq;		/* dump frequency in days */
+	int mnt_passno;		/* pass number on parallel fsck */
```

Comment:
I feel like it is a mistake to replicate the `mntent` structure and think the `sc_mount_entry` struct should look like this:

```c
struct sc_mount_entry {
	struct mntent entry;

	struct sc_mount_entry *next;
};
```
My reasoning is that you won't need to convert back-and-forth between `sc_mount_entry` and `mntent` structs. You'll still need to duplicate the strings in `sc_clone_mount_entry_from_mntent()` but `sc_save_mount_profile()` would get smaller because you have a `mntent` ready to be passed to `addmntent()`.

There may come a time that you want to make use of `hasmntopt()` and that'll be simple because you have a `mntent` ready to pass to it.